static/js/tools/choropleth/choropleth.tsx

```diff
@@ -450,6 +542,45 @@ class ChoroplethMap extends Component<PropsType, StateType> {
     }
   };
 
+  /**
+   * Returns the complete select/dropdown component
+   * with the available dates as options.
+   */
+  public datePicker = (): JSX.Element => {
```

Comment:
Can this be made into a new component? Or the computation to get the set of displayed dates should be done only once - it will now happen on each render.

Reply:
It's now its own component that takes 3 props: dates, pickedDate and onSelectedDateChanged.